docker/configure_workers_and_start.py

```diff
@@ -389,29 +554,219 @@ def generate_base_homeserver_config() -> None:
+            log(
+                f"Worker name request found: '{requested_worker_name}'"
+                f", for: {worker_types_set}"
+            )
+
+        else:
+            # The worker name will be the worker_type, however if spaces exist
+            # between concatenated worker_types and the "+" because of readability,
+            # it will error on startup. Recombine worker_types without spaces and log.
+            # Allows for human readability while declaring a complex worker type, e.g.
+            # 'event_persister + federation_reader + federation_sender + pusher'
+            if (len(worker_types_set) > 1) and (" " in worker_base_name):
+                worker_base_name = "+".join(sorted(worker_types_set))
+                log(
+                    "Default worker name would have contained spaces, which is not "
+                    f"allowed: '{worker_type_string}'. Reformed name to not contain "
+                    f"spaces: '{worker_base_name}'"
+                )
+
+        # At this point, we have:
+        #   worker_base_name which might be identical to
+        #   new_worker_type_string which might still be what it was when it came in
+        #   worker_types_set which is a Set of what worker_types are requested
+
+        # Uncommon mistake that will cause problems. Name string containing quotes
+        # or spaces will do Bad Things to filenames and probably nginx too.
+        if (
+            (" " in worker_base_name)
+            or ('"' in worker_base_name)
+            or ("'" in worker_base_name)
+        ):
+            error(
+                "Requesting a worker name containing a quote mark or a space is "
+                "not allowed, as it would raise a FileNotFoundError. Please fix: "
+                f"{worker_base_name}"
+            )
+
+        # This counter is used for naming workers with an incrementing number. Use the
+        # worker_base_name for the index
+        worker_type_counter[worker_base_name] += 1
+
+        # Name workers by their type or requested name concatenated with an
+        # incrementing number. e.g. federation_reader1 or event_creator+event_persister1
+        worker_name = worker_base_name + str(worker_type_counter[worker_base_name])
+
+        # Now that the worker name is settled, check this name isn't used for a
+        # different worker_type. If it's not allowed, will error and stop. If no
+        # issues, it will be added to the counter. This will prevent accidentally
+        # naming a worker by a worker_type. e.g. 'pusher, pusher=user_dir'
+        # Make sure the worker types being checked are deterministic.
+        deterministic_worker_type_string = "+".join(sorted(worker_types_set))
+        check_worker_type = worker_name_checklist.get(worker_base_name)
+        # Either this doesn't exist yet, or it matches with a twin
+        if (check_worker_type is None) or (
+            check_worker_type == deterministic_worker_type_string
+        ):
```

Comment:
`worker_base_name` is still useful in the future, for load-balancing in nginx(if more than 1). Before, a worker was named after it's type, now it can have something custom which increases the complexity.

Reply:
Plus, this nearing the bottom of this giant `for` loop, and we might still be looking for more workers that don't make the cut.

Reply:
Have to still check for `None` here, as it may not have been seen yet. Updated some things around this in e2fecab See if that makes more sense?